internal/eea/eea.go

```diff
@@ -311,58 +285,53 @@ func (e *EEA) buildEntityWrapper(
 
 func (e *EEA) buildRepositoryInfoWrapper(
 	ctx context.Context,
-	repoID uuid.NullUUID,
+	repoID uuid.UUID,
 	projID uuid.UUID,
 ) (*entities.EntityInfoWrapper, error) {
-	providerID, r, err := getRepository(ctx, e.querier, projID, repoID.UUID)
+	providerID, r, err := getRepository(ctx, e.querier, projID, repoID)
 	if err != nil {
 		return nil, fmt.Errorf("error getting repository: %w", err)
 	}
 
 	return entities.NewEntityInfoWrapper().
 		WithRepository(r).
-		WithRepositoryID(repoID.UUID).
+		WithRepositoryID(repoID).
 		WithProjectID(projID).
 		WithProviderID(providerID), nil
 }
 
 func (e *EEA) buildArtifactInfoWrapper(
 	ctx context.Context,
-	repoID uuid.NullUUID,
+	artID uuid.UUID,
 	projID uuid.UUID,
-	artID uuid.NullUUID,
 ) (*entities.EntityInfoWrapper, error) {
-	providerID, a, err := artifacts.GetArtifact(ctx, e.querier, projID, artID.UUID)
+	providerID, a, err := artifacts.GetArtifact(ctx, e.querier, projID, artID)
 	if err != nil {
 		return nil, fmt.Errorf("error getting artifact with versions: %w", err)
 	}
 
 	eiw := entities.NewEntityInfoWrapper().
 		WithProjectID(projID).
 		WithArtifact(a).
-		WithArtifactID(artID.UUID).
+		WithArtifactID(artID).
 		WithProviderID(providerID)
-	if repoID.Valid {
-		eiw.WithRepositoryID(repoID.UUID)
-	}
 	return eiw, nil
 }
 
 func (e *EEA) buildPullRequestInfoWrapper(
 	ctx context.Context,
-	repoID uuid.NullUUID,
+	prID uuid.UUID,
 	projID uuid.UUID,
-	prID uuid.NullUUID,
 ) (*entities.EntityInfoWrapper, error) {
-	providerID, pr, err := getPullRequest(ctx, e.querier, projID, repoID.UUID, prID.UUID)
+	providerID, repoID, pr, err := getPullRequest(ctx, e.querier, projID, prID)
 	if err != nil {
 		return nil, fmt.Errorf("error getting pull request: %w", err)
 	}
 
 	return entities.NewEntityInfoWrapper().
-		WithRepositoryID(repoID.UUID).
+		WithRepositoryID(repoID).
 		WithProjectID(projID).
 		WithPullRequest(pr).
-		WithPullRequestID(prID.UUID).
+		WithPullRequestID(prID).
 		WithProviderID(providerID), nil
 }
```

Comment:
**nit:** it feels that code might get shorter both here and inside `getPullRequest` (and the others) if you pass `eiw` down to it instead of returning data to add.